io/net/htx_nic_devices.py

```diff
@@ -189,6 +189,8 @@ def build_htx(self):
             if host_distro_pattern == peer_distro_pattern:
                 if process.system(cmd, shell=True, ignore_status=True):
                     self.cancel("Installion of rpm failed")
+                if not self.session.connect():
+                    self.cancel("failed connecting to peer")
```

Comment:
if no connection.. should not you connect the session instead of failing ?

Reply:
@vaishnavibhat any update here?